### DISCUSSION TITLE
RSC: Nextjs.14 dehydration with prefetch problems

### DISCUSSION THREAD

**Discussion by igortas** [TanStack/query#6400] [Q&A]
Hi,

I've trying to migrate ```CRA``` app to ```Next.JS``` with ```app``` router, but I'm losing the boundary for ```RQ``` inside the server components after two sequential ```redirects```.

When I'm redirected to first subpath from ```/``` to ```/path1``` I'm trying to prefetch query and dehydrate the cache inside the server components in Next.JS 14 but it throws ```json``` syntax error: ```unhandledRejection: SyntaxError: Unexpected token '<', "<!DOCTYPE "... is not valid JSON``` when running: ```req.json()```

This is weird because no problem with the same query on the client side exists, it's just ```GET``` endpoint, but the second ```prefetch``` query works without problem.

BR

**Comment by TkDodo** [comment]
I need to see a codesandbox with that error before I can say anything about that
